pgmpy/inference/ExactInference.py

```diff
@@ -53,6 +53,8 @@ def _get_working_factors(self, evidence):
         # Dealing with evidence. Reducing factors over it before VE is run.
         if evidence:
             for evidence_var in evidence:
+                if evidence_var not in working_factors:
```

Comment:
I think this should go with other checks in the `query` and `map_query` methods.